python/ql/test/query-tests/Security/CWE-798/test.py

```diff
@@ -17,3 +17,14 @@ def sell(client, units):
     conn.cmd("sell", 1000)
     conn.close()
 
+
+# Ignored
+test(password='short')
+test(password='Capitalized')
+test(password='  whitespace') # too much whitespace
+test(password='insecure__') # too many underscores
+test(password='aaaaaaaaaa') # too repetitive
+test(password='format_string_{}')
+
+# TODO: we think this is a format string :\
+test(password='''U]E8FPETCS_]{,y>bgyzh^$yC5>SP{E*2=`;3]G~k&+;khy3}4]jdpu;D(aP$SCFA{;hh4n46pUJ%+$nEP_gqNq#X!2$%*C-6y6%''')
```

Comment:
And this means we do not consider it to be a string? That sounds wrong to me.

Reply:
No, it's just that if it's a format string, then we don't consider it to be a credentials https://github.com/Semmle/ql/blob/c1d073a54dd299393fecc550b594c4a973278dbf/python/ql/src/Security/CWE-798/HardcodedCredentials.ql#L69

Reply:
Ah, this makes sense. We wouldn't want to flag random format strings as containing credentials (though I wonder how often this happens. We might want to see what the impact of removing `not format_string(str)` is on LGTM).